### PR TITLE
Add progress bar

### PR DESCRIPTION
**Pull request by Nipher**
https://user-images.githubusercontent.com/7523828/189809698-ab9ab090-eaa3-43e4-a2bb-c8bb71cf7e30.mp4

### CODE REVIEW

**Review by swemail — APPROVED** [review]
Nice work!

File: client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  index?: number;
+};
+
+const ProgressBar: React.FC<ProgressBarProps> = ({index = 0, length = 1}) => (
+  <Wrapper>
+    {Array(length - 1)
+      .fill('')
+      .map((_, idx) => (
+        <Block key={idx} fill={idx < index} />
+      ))}
```
Comment (gewfy) [inline client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx:61]: Out of curiosity. Any particular reason to chink this up in blocks? One bar with with a calculated percentage?

File: client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const style = useAnimatedStyle(() => ({
+    width: width.value,
+  }));
+
+  useEffect(() => {
+    width.value = withTiming(fill ? '100%' : '0%', {
+      duration: 500,
+      easing: Easing.bezier(0.33, 1, 0.68, 1),
+    });
+  }, [width, fill]);
```
Comment (gewfy) [inline client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx:41]: Nice with animation :)

**Review by gewfy — APPROVED** [review]
Nice!!

File: client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (Nipher) [reply client/src/routes/Temple/components/ProgressBar/ProgressBar.tsx:61]: felt simpler with blocks to me

**Pull request merged**